src/webgpu/shader/execution/expression/call/builtin/atomics/atomicOr.spec.ts

```diff
@@ -0,0 +1,53 @@
+export const description = `
+Atomically read, or and store value.
+
+* Load the original value pointed to by atomic_ptr.
+* Obtains a new value by or'ing with the value v.
+* Store the new value using atomic_ptr.
+
+Returns the original value stored in the atomic object.
+`;
+
+import { makeTestGroup } from '../../../../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../../../../gpu_test.js';
+
+import { dispatchSizes, workgroupSizes, runTest } from './harness.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('or')
+  .specURL('https://www.w3.org/TR/WGSL/#atomic-rmw')
+  .desc(
+    `
+AS is storage or workgroup
+T is i32 or u32
+
+fn atomicOr(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
+`
+  )
+  .params(u => u.combine('workgroupSize', workgroupSizes).combine('dispatchSize', dispatchSizes))
+  .fn(t => {
+    const numInvocations = t.params.workgroupSize * t.params.dispatchSize;
+
+    // Allocate an output buffer with bitsize of max invocations plus 1 for validation
+    const bufferNumElements = Math.max(1, numInvocations / 32) + 1;
+
+    // Start with all bits low, then using atomicOr to set global id bit on.
+    // Note: Both WGSL and JS will shift left 1 by id modulo 32.
+    const initValue = 0;
+    const op = `atomicOr(&output[id / 32], 1u << id)`;
```

Comment:
This is just going to output `0xffffffff` for most atomics, right?

I think it would be preferable to emit something less regular. Maybe you could have a black-box function that takes an `id` and returns a new integer `i`: `let i = f(id);`.
`f` could then be passthrough like it is now, or something like `(id * large_prime) % count`

Reply:
> This is just going to output 0xffffffff for most atomics, right?
Sort of. The test combines workgroupSize and dispatchSize which contain values like 1, 2, 4, etc.:
```
export const workgroupSizes = [1, 2, 32, 64];
export const dispatchSizes = [1, 4, 8, 16];
```
So when, say, 2 * 1 are tested, then only the first two bits are going to be set.

Do you still think the blackbox function is necessary? If I do add it, then should this become yet another parameter to the tests?

Reply:
Just pushed a commit that adds this feature, used by the atomic bit ops (atomicAnd/Or/Xor). Let me know if this is what you were expecting.